src/exchange/group/accounts/group-accounts.controller.js

```diff
@@ -79,17 +92,53 @@ angular.module('Module.exchange.controllers').controller(
       }
     }
 
+    getDefaultDomain() {
+      const name = _.last(this.currentAccount.split('@'))
+      return {
+        displayName: name,
+        name: name
+      };
+    }
+
+    fetchAccountCreationOptions() {
+      this.loadingDomains = true;
+      return this.services.Exchange.fetchingAccountCreationOptions(
+        this.$routerParams.organization,
+        this.$routerParams.productId,
+      )
+        .then((accountCreationOptions) => {
+          const domains = [this.allDomainsOption];
+          _.forEach(accountCreationOptions.availableDomains, domain => {
```

Comment:
`forEach` + `push` === `map`

Reply:
Yes, I agree with you. In this case, I need to add elements to an existing array. map creates a new array. To add them into the same array, I need to runt he loop again. Let me know if I got it wrong :)

Reply:
You are absolutely right, so you could do something like that:
```js
const defaultDomain = [this.allDomainsOption];
const availableDomains = accountCreationOptions.availableDomains.map(domain => {
    if (domain.name === this.selectedDomain.name) {
        this.selectedDomain = domain;
    }

    return domain;
});

this.availableDomains = [defaultDomain, ...availableDomains];
```

Reply:
```
const domains = [this.allDomainsOption];
          _.forEach(accountCreationOptions.availableDomains, domain => {
            domains.push(domain);
            if (domain.name === this.selectedDomain.name) {
              this.selectedDomain = domain;
            }
          });
          this.availableDomains = domains;
```

This is the code I am using. It is same as what you have mentioned above. Should I change it or retain it?